### PR TITLE
Add existing_import API

### LINKED ISSUE #1439: Implement `/existing_import` API endpoint.
## Problem
We want to implement an `existing_import` API endpoint that will be exposed to the frontend which will allow us to move/import records from the temp table to an existing table. 
  
## Proposed Solution
The URL for the endpoint should look something like: `api/db/vX/tables/{temp_table_id}/existing_import/` and the request payload should look like this:
### Request
```json
{
	"table_to_import_to": "<existing_table_id>",
	"data-files":[10],
	"mappings":{
	    "id": "id",
            "Name": "Center",
            "Status": "Status",
            "Case_id:": "Case_Number"
        }
}

```

### PR DESCRIPTION
**Pull request by Anish9901**
Fixes #1439 

**Technical details**

This PR is the implementation of the `existing_import` API endpoint which will allow us to import records from one table to another with a request like this:
### Request:
```json
{
	"table_to_import_to": "<existing_table_id>",
	"data-files":[10],
	"mappings":{
	    "id": "id",
            "Name": "Center",
            "Status": "Status",
            "Case_id:": "Case_Number"
        }
}
```

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: mathesar/api/db/viewsets/tables.py [hunk 1]
```diff
@@ -146,3 +148,28 @@ def previews(self, request, pk=None):
...
+        serializer.is_valid(raise_exception=True)
+        existing_table_id = serializer.validated_data['table_to_import_to']
+        data_files = serializer.validated_data['data-files']
+        mappings = serializer.validated_data['mappings']
+        existing_table = get_table_or_404(existing_table_id)
+
+        try:
+            insert_records_to_existing_table(
+                existing_table, temp_table, mappings, data_files
+            )
```
Comment (Anish9901) [inline mathesar/api/db/viewsets/tables.py:165]: @mathemancer  Should we make this a method under the `Table` model?

Comment (mathemancer) [reply mathesar/api/db/viewsets/tables.py:165]: Yes, I think that would be good.

**Review by mathemancer — CHANGES_REQUESTED** [review]
So, this is a start. I have a change to request that I should have caught earlier:

This PR seems to assume we're going to need the data file for some reason at this stage. Why would that be? I think this should be a general endpoint that lets one simply take the records from _any_ table A, insert them to table B, and then (optionally) drop table A.

**Comment by Anish9901** [comment]
> This PR seems to assume we're going to need the data file for some reason at this stage. Why would that be? I think this should be a general endpoint that lets one simply take the records from _any_ table A, insert them to table B, and then (optionally) drop table A.

I agree, the data_files are only there to correct the `data_file.table_imported_to` field. Since the file was imported to Table A the `table_imported_to` field will point to Table A, but once we drop Table A, `table_imported_to` should point to Table B.

**Pull request merged**